include/nuttx/note/note_driver.h

```diff
@@ -62,4 +62,74 @@ int note_initialize(void);
 
 #endif /* defined(__KERNEL__) || defined(CONFIG_BUILD_FLAT) */
 
+#if CONFIG_DRIVER_NOTERAM_TASKNAME_BUFSIZE > 0
+
+/****************************************************************************
+ * Name: note_get_taskname
+ *
+ * Description:
+ *   Get the task name string of the specified PID
+ *
+ * Input Parameters:
+ *   PID - Task ID
+ *
+ * Returned Value:
+ *   Pointer to the task name string
+ *   If the corresponding name doesn't exist in the buffer, NULL is returned.
+ *
+ ****************************************************************************/
+
+const char *note_get_taskname(pid_t pid);
```

Comment:
```suggestion
FAR const char *note_get_taskname(pid_t pid);
```

Reply:
not fix

Reply:
```suggestion
int note_get_taskname(pid_t pid, FAR char *name);
```
let's copy the internal buffer instead return raw pointer to avoid the race condition